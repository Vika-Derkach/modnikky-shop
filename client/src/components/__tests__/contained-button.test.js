//import "jest-dom/extend-expect";
import "@testing-library/jest-dom/extend-expect";
import { cleanup, render } from "@testing-library/react";
import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import ContainedButton from "../contained-button";

afterEach(() => {
  cleanup();
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<ContainedButton></ContainedButton>, div);
});

it("renders button correctly", () => {
  const { getByTestId } = render(<ContainedButton button_text="hi world" />);
  const newButton = getByTestId("contained-button");

  expect(newButton).toBeInTheDocument();
  expect(newButton).toHaveTextContent("hi world");
  expect(newButton).not.toContainHTML("strike");
});
it("matches snapshot 1", () => {
  const tree = renderer.create(<ContainedButton button_text="save" />).toJSON();
  expect(tree).toMatchSnapshot();
});
it("matches snapshot 2", () => {
  const tree = renderer
    .create(<ContainedButton button_text="click me" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
