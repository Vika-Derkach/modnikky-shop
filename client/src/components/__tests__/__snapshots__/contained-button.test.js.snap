// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <button
    className="button"
    data-testid="contained-button"
  >
    save
  </button>
</div>
`;

exports[`matches snapshot 1 1`] = `
<div>
  <button
    className="button"
    data-testid="contained-button"
  >
    save
  </button>
</div>
`;

exports[`matches snapshot 2 1`] = `
<div>
  <button
    className="button"
    data-testid="contained-button"
  >
    click me
  </button>
</div>
`;
